bot/seasons/evergreen/battleship.py

```diff
@@ -0,0 +1,387 @@
+                break
+
+            square.aimed = True
+            await turn_message.delete()
+            for message in alert_messages:
+                await message.delete()
+
+            alert_messages = []
+            alert_messages.append(await self.next.send(f"{self.turn} aimed at {self.match.string}!"))
+
+            if square.boat:
+                await self.turn.send("Hit!", delete_after=3.0)
+                alert_messages.append(await self.next.send("Hit!"))
+                if self.check_sink(self.grids[self.next], square.boat):
+                    await self.turn.send(f"You've sunk their {square.boat} ship!", delete_after=3.0)
+                    alert_messages.append(await self.next.send(f"Oh no! Your {square.boat} ship sunk!"))
+                    if self.check_gameover(self.grids[self.next]):
+                        await self.turn.send("You win!")
+                        await self.next.send("You lose!")
+                        self.gameover = True
+                        await self.game_over(winner=self.turn, loser=self.next)
+                        break
+            else:
+                await self.turn.send("Miss!", delete_after=3.0)
+                alert_messages.append(await self.next.send("Miss!"))
+
+            self.turn, self.next = self.next, self.turn
+
+
+class Battleship(commands.Cog):
+    """Play the classic game Battleships!"""
+
+    def __init__(self, bot: commands.Bot) -> None:
+        self.bot = bot
+        self.games: typing.List[Game] = []
+        self.waiting: typing.List[discord.Member] = []
+
+    def already_playing(self, player: discord.Member) -> bool:
+        """Check if someone is already in a game."""
+        return player in [getattr(game, x) for game in self.games for x in ("player1", "player2")]
+
+    @commands.group(invoke_without_command=True)
+    @commands.guild_only()
+    async def battleship(self, ctx: commands.Context) -> None:
+        """
+        Play a game of Battleships with someone else!
+
+        This will set up a message waiting for someone else to react and play along.
+        The game takes place entirely in DMs.
+        Make sure you have your DMs open so that the bot can message you.
+        """
+        if self.already_playing(ctx.author):
+            return await ctx.send("You're already playing a game!")
+
+        if ctx.author in self.waiting:
+            return await ctx.send("You've already sent out a request for a player 2")
+
+        announcement = await ctx.send(
+            "**Battleships**: A new game is about to start!\n"
+            f"Press :raising_hand: to play against {ctx.author.mention}!"
+        )
+        self.waiting.append(ctx.author)
+        await announcement.add_reaction("\U0001f64b")
+
+        def predicate(reaction: discord.Reaction, user: discord.Member):
+            if self.already_playing(ctx.author):  # If they've joined a game since requesting a player 2
+                raise asyncio.TimeoutError
```

Comment:
I think this can still be an issue. The game's author could join another game in the time between lines 366 and 370. I do think that if the author attempts to join another game while their own game is still waiting, the author should be asked to explicitly cancel their own game first and then re-attempt to join the other game.